### ISSUE TITLE
Recommend using dataclass in "Classes as dependencies" Docs

### ISSUE THREAD

**Issue by johnthagen** [tiangolo/fastapi#3114]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * Or, I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

## Idea

Currently in the Classes as Dependencies docs: https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/#classes-as-dependencies_1

The sample class is:

```py
class CommonQueryParams:
    def __init__(self, q: Optional[str] = None, skip: int = 0, limit: int = 100):
        self.q = q
        self.skip = skip
        self.limit = limit
```

I believe that it would be better to recommend the cleaner use of a `dataclass`, to remove boilerplate that can be error prone. New users are likely to simply copy this style directly and not realize they can use a cleaner syntax. It also matches more similarly to Pydantic usage elsewhere in FastAPI:

```py
from dataclasses import dataclass

@dataclass
class CommonQueryParams:
    q: Optional[str] = None
    skip: int = 0
    limit: int = 100

@app.get("/items/")
async def read_items(commons: CommonQueryParams = Depends(CommonQueryParams)):
    response = {}
    if commons.q:
        response.update({"q": commons.q})
    items = fake_items_db[commons.skip : commons.skip + commons.limit]
    response.update({"items": items})
    return response
```

This is somewhat discussed in the closed issue #318 from two years ago.

**Comment by falkben** [comment]
FastAPI is compatible with Python 3.6+. As dataclasses were added in 3.7, my guess is that it was just simpler to document the version that works for both of them. I suppose a snippet could be added to the docs that mentioned that if using 3.7+, you could use dataclasses

**Comment by johnthagen** [comment]
Good point. It's also possible to use `dataclasses` on 3.6 using the [PyPI backport](https://pypi.org/project/dataclasses/) as well.

**Comment by tiangolo** [comment]
Thanks for the interest and effort!

I'll copy my answer to the PR here as well for completeness.

---

I prefer to leave it as is, because dataclasses require assuming some knowledge about dataclasses and how they work. And they would only work for some specific cases. They wouldn't necessarily take things like `Query()`, `Body()` as default values, or at least the behavior is not certain. And it would require explaining how they work.

I prefer to keep the explanation simple and pure, that makes it obvious that the thing that makes it callable is what would be used to get the information (in this case the `__init__()`. But in a dataclass that is hidden, so it would require explaining that it's hidden but created automatically, and then it would require explaining that because the `__init__()` is autocreated, some things would work and some wouldn't.

And although the code is shorter, the explanation and the cognitive load would be higher, explaining what's everything that is happening underneath.

So I'll pass on this suggestion, but thanks for the interest! :cake:
